scripts/text_generation/index.rst

```diff
@@ -6,7 +6,80 @@ Text Generation
 Sampling a Language Model
 +++++++++++++++++++++++++
 
-This script can be used to generate sentences using beam search or a sequence sampler, to sample from a pre-trained language model.
+This script can be used to generate sentences using beam search or a sequence sampler, to sample from a pre-trained language model such as GPT-2. For example:
+
+.. code-block:: console
+
+   $ python sequence_sampling.py random-sample \
+         --bos 'Deep learning and natural language processing' \
+         --beam-size 1 --print-num 1 \
+         --lm-model gpt2_345m \
+         --max-length 1024
+
+Output is
+
+.. code-block:: console
+
+    Sampling Parameters: beam_size=1, temperature=1.0, use_top_k=None
+    Generation Result:
+    ['Deep learning and natural language processing brought application choice in healthcare and perception of sounds and heat to new heights, enriching our physical communities with medical devices and creating vibrant cultures. Anecdote is slowly diminishing but is hardly obsolete nor more appealing than experience.Despite those last words of wisdom, most headset makers even spook us with the complexity and poor code quality. the hard set a mere $150 and beginner creates center for getting started. Temp cheap:\nPosted by Fleegu at 12:02 PM<|endoftext|>', -461.15128]
+
+Sequence Sampler
+~~~~~~~~~~~~~~~~
+
+Use the following command to decode to sample from the multinomial distribution.
+
+.. code-block:: console
+
+   $ python sequence_sampling.py random-sample --bos 'I love it' --beam-size 5 --print-num 5
+
+Output is
+
+.. code-block:: console
+
+   Sampling Parameters: beam_size=5, temperature=1.0, use_top_k=None
+   Generation Result:
+   [u'I love it and martial arts , history , and communism ; it is seems to be probably a date .', -76.772766]
```

Comment:
Why does this result looks so much different from the previous one? for example, spaces around comma/period. And removing the unicode notation helps improve readability `[u'I love it in all ` -> `['I love it in all `

Reply:
results seem updated. since I changed it to use a proper (sacremoses) tokenizer and detokenizer, and also changed it to end according to model's `<eos>`, I updated all the results again.